json-rpc/src/data.rs

```diff
@@ -291,3 +293,25 @@ pub fn get_account_state_with_proof(
         account_state_with_proof,
     )?)
 }
+
+/// Get all resources stored under `account_address` at `version`
+pub fn get_resources(
+    db: &dyn MoveDbReader,
+    ledger_version: u64,
+    account_address: AccountAddress,
+    version: u64,
+) -> Result<BTreeMap<String, AnnotatedMoveStruct>, JsonRpcError> {
```

Comment:
I wonder why this function does not call to new function created for get_resource, instead of doing same thing by calling get_account_state_with_proof

Reply:
Ah, it's because `get_resource` fetches the entire state blob + proof and then returns a single resource. I didn't want to fetch the state blob N times for an account with N resources, so I wrote slightly more code here instead of reusing `get_resource`.